src/ray/core_worker/transport/thread_pool.cc

```diff
@@ -14,21 +14,57 @@
 
 #include "ray/core_worker/transport/thread_pool.h"
 
+#include <boost/asio/post.hpp>
+#include <future>
 #include <memory>
+#include <utility>
 
 namespace ray {
 namespace core {
 
-BoundedExecutor::BoundedExecutor(int max_concurrency) {
+BoundedExecutor::BoundedExecutor(
+    int max_concurrency,
+    std::function<std::function<void()>()> initialize_thread_callback)
```

Comment:
Let's add cpp unit tests to check the `initialize_thread_callback` behavior:

1. It's called when constructing the thread pool
2. The releaser is called when joining

Reply:
updated